internal/characters/xiao/xiao.go

```diff
@@ -0,0 +1,128 @@
+package xiao
+
+import (
+	"github.com/genshinsim/gsim/pkg/core"
+	"github.com/genshinsim/gsim/pkg/character"
+)
+
+func init() {
+	core.RegisterCharFunc("xiao", NewChar)
+}
+
+// Xiao specific character implementation
+type char struct {
+	*character.Tmpl
+	eCharge      int
+	eChargeMax   int
+	eNextRecover int
+	eTickSrc     int
+	qStarted     int
+	a4Expiry     int
+	c6Check      []interface{} // int (source frame), string (snapshot ability name)
```

Comment:
Shouldn't use interface here as that's actually slower. This should just be

`c6Src int`

Reply:
also need a counter here
`c6Count int`